omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/FrontendJsonConfigController1_9.java

```diff
@@ -0,0 +1,128 @@
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs1_9;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.charset.StandardCharsets;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.openmrs.User;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+import org.openmrs.util.OpenmrsUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.apache.commons.io.IOUtils;
+import org.codehaus.jackson.JsonProcessingException;
+import org.codehaus.jackson.map.ObjectMapper;
+
+@Controller
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/frontend/config.json")
+public class FrontendJsonConfigController1_9 extends BaseRestController {
+
+    private static final String DEFAULT_FRONTEND_DIRECTORY = "frontend";
+    private static final String GP_LOCAL_DIRECTORY = "spa.local.directory";
+    private static final String JSON_CONFIG_FILE_NAME = "config.json";
+    private static final Logger log = LoggerFactory.getLogger(FrontendJsonConfigController1_9.class);
+
+    @RequestMapping(method = RequestMethod.GET)
+	public void getFrontendConfigFile(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        File jsonConfigFile = getJsonConfigFile();
+        if (!jsonConfigFile.exists()) {
+            log.debug("Configuration file does not exist");
+            response.sendError(HttpServletResponse.SC_NOT_FOUND, "Configuration file does not exist");
+        }
+        try {
+            InputStream inputStream = new FileInputStream(jsonConfigFile);
+            OutputStream outStream = response.getOutputStream();
+            OpenmrsUtil.copyFile(inputStream, outStream);
+    
+            response.setContentType("application/json");
+            response.setHeader("Content-Disposition", "attachment; filename=" + jsonConfigFile.getName());
```

Comment:
Adding `Content-Disposition: attachment` makes it impossible to use this on the frontend because you're not returning the JSON file as a REST response. You're returning a response instructing the user agent to download that as a file. Without some kind of endpoint that allows us to make a get request and get the JSON document back as part of the response without the `Content-Disposition` we cannot make use of this configuration file. (I'd suggest adding an optional parameter, e.g., `?download` so that:

`GET /openmrs/ws/rest/v1/frontend/config.json` returns like requesting any other JSON file and `GET /openmrs/ws/rest/v1/frontend/config.json?download` sends it as an attachment.

Reply:
i have added the `Content-Disposition: attachment` for both `GET /openmrs/ws/rest/v1/frontend/config.json?download` and `GET /openmrs/ws/rest/v1/frontend/config.json?download=true`.  As an addition, i have Set their content type to `application/x-download`
Thank you for this clarification

Reply:
I'd keep the content type! It's still a JSON document.